### ISSUE TITLE
Research Reporting Support

### ISSUE THREAD

**Issue by fortepc** [PokeNavBot/issue-tracker#140]
It would be cool if there was an announce research here option exactly like the announce raids here. 

The process could be the bot recognizes the screenshots scans it for the pokestop name, then asks the user what type of reward it is, then ask e user what the research task is.  

As a stretch goal, it could also allow the reporting of a nearby Pokemon. So instead of the bot recognizes the pokestop and then asks what the research reward is, it asks what they're reporting either a research or a nearby spawn.

**Comment by adetia9** [comment]
Left multiple comments in #feature requests today, regarding organizing a research summary that would sort them by task, etc, into a very orderly list.  can provide more info if desired, past what i put in the channel.

**Comment by dandesousa** [comment]
@adetia9 please paste what you put in the ticket, as it would really help me to keep everything in one place

**Comment by dandesousa** [comment]
Additional Notes: https://github.com/PokeNavBot/Issue-Tracker/issues/181#issue-370929649

**Comment by Rhendak** [comment]
![image](https://user-images.githubusercontent.com/1855295/47081181-337aec80-d20a-11e8-8b3a-cc5e1ade1b05.png)

Saw this image being referenced in the Discord in the #feature-request channel. Adding this for more information.

**Comment by adetia9** [comment]
Some additional information, the research system mentioned above lets you do a $research and then prompts you to fill in task (which you can abbreviate, it will ask you to pick from a pool if it needs to), pokestop, etc.  or you can do it all in one shot with all the text entered in one line.  i think having both options is awesome.

i like the above system, because it sorts by tasks, is easily visually to see what you need, and keeps the whole channel tidy, clear it out at midnight and start anew the next day.   

they have also included !wild reporting in that same summary. (screenshots and stealing their ideas with their blessing from tehstone who works on the server shown in these screenies).  This is a great value added service, but the research summary without it is something spectacular on its own.

also, because we as players are competitive, they introduced a leaderboard for reporters, which i personally love, but, again, takes a second seat to the system itself.
![res1](https://user-images.githubusercontent.com/10247739/47230737-49aeb700-d388-11e8-9757-f6ffd039de3c.JPG)

**Comment by Rhendak** [comment]
To add to the above comment, i think leaderboards for research should definitely be included (we have badges in our server depending on this). However for the initial release the $research function would be most important. Leaderboards could be added in a later stage.

**Comment by buneech** [comment]
I believe this as a logical step after poi submissions, as the pokestops can now be recognised and have correct directions to it.

Something like Rhendak has attached would be perfect, a summary of research tasks reported for that day.
I think it is from a fork of Meowth bot, and it supports research reporting using a one liner command or interactively, question based for task, reward and pokestop. I see the interactive part as a bonus, but I don't think it would be necessary, as the raid and egg reporting commands are also easy enough for submissions.

**Comment by Rhendak** [comment]
**Chat log from 9-11-2018**

IVpipsLast Friday at 6:17 PM
I do think that since research doesn't have a coordination component to it
the way its published can be different
for example, unlike raids and trades ,there really is no interaction -- its a bulletin board, with an option to delete research
Don't need separate messages
I do want to make research as easy as possible to report
zFatKatLast Friday at 6:18 PM
Right now we are using the stop info command in area channels and it’s overwhelming
IVpipsLast Friday at 6:18 PM
So I think a pokestop screenshot grab
zFatKatLast Friday at 6:18 PM
Trying to get people to use a standard format
Ugh.  Most of it is just random uploads
IVpipsLast Friday at 6:19 PM
then an interactive prompts or a screen grab plus the reward
a siri shortcut could make this really easy, or something in the app that proxies the report
RhendakLast Friday at 6:19 PM
Well I was thinking the other day @IVpips. It would be kinda cool if people could confirm that the quest is on that pokestop by reacting with :thumbsup: or deny it with :thumbsdown:. 5+ :thumbsdown: would remove it from the list?
IVpipsLast Friday at 6:19 PM
yeah thats a good idea
that requires one message per report
which is likely fine
zFatKatLast Friday at 6:20 PM
Work your magic lol
RhendakLast Friday at 6:20 PM
To keep the channel clean I would prefer doing it with a command, that is afterwards deleted by the bot.
command + interactive promts that is
zFatKatLast Friday at 6:20 PM
A map really isn’t needed but a clean list would be great
IVpipsLast Friday at 6:21 PM
btw while I'm fine with discussing it here, theres like a 50% chance I'll forget what we talked about so anyone feel free to drop any notes from the discussion on the github issues -- its helps me enormously to collate all thoughts
RhendakLast Friday at 6:22 PM
Now that the POI system is (kinda) in place, that kinda puts down the foundation for the research functionality
IVpipsLast Friday at 6:23 PM
Yeah POI is at an MVP (minimal viable product) stage(edited)
just enough to unblock almost everyone
zFatKatLast Friday at 6:30 PM
So far everything has made my life a lot easier as far as organizing goes.  The only thing that we don’t have organized well is a quest data base.
RhendakLast Friday at 6:30 PM
Same here @zFatKat
zFatKatLast Friday at 6:30 PM
I am still cross posting a lot of info bc there is no simple way to do it
IVpipsLast Friday at 6:30 PM
Yeah so as with many features, if suddenly someone mentions something or I have an epiphany on a great, clean way to do it, I will basically drop everything and have quests in PokeNav in a few days
then in the app after
IVpipsLast Friday at 6:31 PM
Thats sort of how trading, poi happened
TL;DR a webhook lets anyone with the webhook address post(edited)
zFatKatLast Friday at 6:35 PM
It could really be as simple as when a pokestop is uploaded the bot asks what quest it is and they press a number
IVpipsLast Friday at 6:35 PM
yeah for the quest I believe that was suggested by forte in the ticket
zFatKatLast Friday at 6:35 PM
Similar to when there are multiple stops with the same name.  Then it could feed to an active quest channel like the raids
It would also stop people from posting a hundred 200 dust quests :rolling_eyes:
Here is your 200 dust quest...  catch two Pokemon
No one is driving across town for 200 dust omg
IVpipsLast Friday at 6:37 PM
how would it stop them?
zFatKatLast Friday at 6:37 PM
If only the top quests are options in the bot
If it were multiple choice so to speak
IVpipsLast Friday at 6:39 PM
so correct me if I'm wrong but all quests can be categorized by the following:

Reward: Encounter (any pokemon theoretically)
Reward: Dust
Reward: Pokeballs
Reward: Berries
Reward: TM
Reward: Other (stones, raid pass etc that are rare)
QualibertLast Friday at 6:39 PM
Just dropping in out of nowhere, because it was not mentionned: I guess multilanguages will still be supported?
IVpipsLast Friday at 6:39 PM
for quests?
QualibertLast Friday at 6:39 PM
Yep
IVpipsLast Friday at 6:39 PM
so thats also on my list (of things I want to work on)(edited)
zFatKatLast Friday at 6:39 PM
That’s the quest list
IVpipsLast Friday at 6:40 PM
bot wide not just quest
QualibertLast Friday at 6:40 PM
Because that would be very hard to get my people to report them if they don't match with their games :frowning:
I suceed to get the older trainers on Discord XD
Like 40-50 and they are a little lost XD
IVpipsLast Friday at 6:41 PM
well i dont think that the bot will read the quest list screenshot, I think if it does screengrab it will be the pokestop screenshot
zFatKatLast Friday at 6:41 PM
My people speak the same language as the bot and still have issues lol
IVpipsLast Friday at 6:41 PM
but thanks for bringing that up
maybe the start of localization is an important dependency
QualibertLast Friday at 6:41 PM
Sure it's just if there a prompt or whatever
Cupkait 🍰Last Friday at 6:59 PM
just popping in to say i really hope quests becomes the next-big-focus after POI, because it seems like the #1 high level feature missing atm.
daadyrbarnLast Friday at 6:59 PM
it is something that's being requested a lot in my server as well :smiley:
IVpipsLast Friday at 7:00 PM
You guys really want quests huh?
🇾
4
🇪
3
🇸
2
Cupkait 🍰Last Friday at 7:00 PM
yes lol
IVpipsLast Friday at 7:00 PM
I'll keep that in mind
Cupkait 🍰Last Friday at 7:00 PM
its the main thing i hear over and over about being lacking as far as integration goes
even though the stop command is a decent workaround
daadyrbarnLast Friday at 7:01 PM
it's  funny because nobody complained about the old system of just posting shots of the stop and research journal, but now everyone is so used to how easy it is with the raid screens :smile:
IVpipsLast Friday at 7:01 PM
yeah I know
Things is you can't get all the information even from multiple screens
I think the simplest and easiest workflow is -> post a stop, select a high level reward
I'm not even necessarily convinced the first pass needs to have what you need to do
In my server, we've been doing a combination of tags + $si
I think I could probably do quests after raid stream but before file upload POI
SunkastLast Friday at 7:04 PM
im surprised more people are interested in research than getting the bot sharding
im not against the research feature, just personally feel sharding is more important
IVpipsLast Friday at 7:04 PM
@Sunkast you probably dont care about bot sharding unless your server uses the bot a lot at peak time
zFatKatLast Friday at 7:04 PM
Quests!  You spoiled my people with Raid Info
Now they want everything to be that easy
Cupkait 🍰Last Friday at 7:05 PM
its not that people odnt care, its that they dont know its an issue for many others. and sharding isnt a feature, i consider it in a sector of its own in comparison
SunkastLast Friday at 7:05 PM
most of our server posts raids between 4 and 7pm EDT
QualibertLast Friday at 7:05 PM
Can't do GPS + Last Quest in journal hun?
QualibertLast Friday at 7:05 PM
Sure it won't work
Just want to know why :P?
IVpipsLast Friday at 7:06 PM
well we're talking about a discord implementation
QualibertLast Friday at 7:06 PM
Oh yeah...won't work without app :frowning:
IVpipsLast Friday at 7:06 PM
The app could do that
but its a totally different flow(edited)
QualibertLast Friday at 7:07 PM
But the OCR could work with ''last quest in journal''?
IVpipsLast Friday at 7:07 PM
theres a lot of blockers for it
IVpipsLast Friday at 7:07 PM
I could identify the last quest in general
it would require image recognition too, like how pokenav identifies egg levels(edited)
Cupkait 🍰Last Friday at 7:08 PM
i dont like that
IVpipsLast Friday at 7:08 PM
most likely its not going to come soon though -- discord first is the PokeNav motto
Cupkait 🍰Last Friday at 7:08 PM
i think it should be primarily tied to the stop itself period because its too easy for someone to just screenshot the last quest then guess the stop wrong
or it be a quest from yesterday and they not realize it, etc.
i also think going based off of a stop screenshot would potentially more easily tie into "is this a research or rare spawn callout?" dialogue to expand eventually into rare spawns also.
IVpipsLast Friday at 7:10 PM
another thing is notifications
that will probably be a follow up
though maybe it isnt AS necessary since they last all day?
QualibertLast Friday at 7:12 PM
I was about to talk about those XD
Well if it's possible, I know some people are looking for special tasks more than raid boss
That pidgey thing to get shinies XD
IVpipsLast Friday at 7:13 PM
Since the infrastructure is in place really its just adding a new condition
"encounter: caterpie"
Cupkait 🍰Last Friday at 7:14 PM
I don’t think notifications are as important just because it is an all day availability. They can check whenever they are playing to see if they can get to it.
zFatKatLast Friday at 7:14 PM
I don’t think notifications for quest or a requirement
IVpipsLast Friday at 7:14 PM
Like most things I'll probably wait for people to request that
Cupkait 🍰Last Friday at 7:14 PM
Value of notifications for quests almost feels like it would be a benefit to spoofers.
Who can go grab whatever whenever.
IVpipsLast Friday at 7:15 PM
I think it could be good as a reminder
zFatKatLast Friday at 7:15 PM
Even if it were available I don’t know if we would implement quest notifications because of the role limits on a server
IVpipsLast Friday at 7:15 PM
but yeah definitely non-essential
Cupkait 🍰Last Friday at 7:15 PM
If you are too lazy to see if it’s available you probably aren’t going to even go get it tbg
Tbh
IVpipsLast Friday at 7:15 PM
man now I wanna work on research :smiley:
zFatKatLast Friday at 7:16 PM
Please
Cupkait 🍰Last Friday at 7:16 PM
:innocent::innocent::innocent:
Goal accomplished
IVpipsLast Friday at 7:17 PM
I've already started the branch
zFatKatLast Friday at 7:33 PM
So about those quests lol
IVpipsLast Friday at 7:35 PM
I'm leaning towards making the research report fail if the stop doesn't exist, any oppositions to that?
zFatKatLast Friday at 7:36 PM
Can I add it later with poi?
IVpipsLast Friday at 7:36 PM
Raids dont fail
thats what im trying to sort out
Cupkait 🍰Last Friday at 7:36 PM
I don’t think it should fail.
IVpipsLast Friday at 7:36 PM
if I force it to be required then I can basically do research is immutable
like trades
meaning theres only one operation on a reearch -- delete it
Cupkait 🍰Last Friday at 7:36 PM
We have tons of stops that weren’t on pgm and were still adding them as we go
zFatKatLast Friday at 7:37 PM
I wouldn’t care if it failed as long as I could see a report of what failed
Cupkait 🍰Last Friday at 7:37 PM
If they fail and don’t post when would we see them
^^^^^^
IVpipsLast Friday at 7:37 PM
good point
zFatKatLast Friday at 7:37 PM
Then I could go behind and add those POY later
IVpipsLast Friday at 7:38 PM
another question, for encounters, require that the encounter pokemon be known
so you can report an encounter of caterpie
Cupkait 🍰Last Friday at 7:39 PM
Yes
IVpipsLast Friday at 7:39 PM
but you can't report an "unknown" encounter
zFatKatLast Friday at 7:39 PM
Agreed
Cupkait 🍰Last Friday at 7:39 PM
Definitely
IVpipsLast Friday at 7:39 PM
i think that important if the "requirements" are optional
Cupkait 🍰Last Friday at 7:39 PM
Agreed
xxTopCatxxLast Friday at 7:46 PM
As I read down, research report to fail if a stop doesn’t exist, could it not be switched to developers notified and research posted with no pinned location and message stating poi needs updating 
encounters Pokemon be known
Agreed but with a buziel did you mean buizel scenario solver would help
And developers I mean server mods /admin/owners- announcement channel :shrug::skin-tone-2:‍♂️
zFatKatLast Friday at 7:55 PM
:point_up_2::point_up_2::point_up_2:
eevvee33Last Friday at 7:56 PM
I know quest announcements (tagging roles) was an important feature on my server.  I wrote  a python program to pull quests from PGM and then report the results to discord, tagging roles if appropriate and adding in the local server emoji (people really liked the emoji matching the reward).
xxTopCatxxLast Friday at 7:56 PM
How about :

Research is posted in one channel that then creates (if not created) and lists it in another channel called daily research(with an option this channel to admin this channel can be deleted daily, around midnight).
Forcing users to post a research to see the research channel every day :shrug::skin-tone-2:‍♂️
zFatKatLast Friday at 7:58 PM
I love that
xxTopCatxxLast Friday at 7:58 PM
If they post the same research then simply replies - this research has already been posted can you confirm it’s blah for blah - thumbs up, thumbs down and allows them to read the daily channel
zFatKatLast Friday at 7:58 PM
Honestly I just want a quest list
eevvee33Last Friday at 7:58 PM
As for unknown encounters, there is always a short list of available encounters, maybe you could print all the possibilities?  "reward: bulbasaur or squirtle or charmander"  Which could later be updated if someone finds out what the actual reward is.
IVpipsLast Friday at 7:59 PM
the first iteration will likely be an MVP that gets it working -- without a known list of research and then I'll see how people respond to it and iterate
👍
1
You want the research channel to be private unless people post a research first :smiley: ?(edited)
Some things I will track as follow up features
xxTopCatxxLast Friday at 8:00 PM
Yeah I think that would be awesome but only a couple of us sat here discussing it
IVpipsLast Friday at 8:00 PM
I think they are good but in order to get it out relatively quickly it will be easier to separates the must have from what can be built after
zFatKatLast Friday at 8:01 PM
I am for the quickest version lol
IVpipsLast Friday at 8:01 PM
You can post it on the github and I'll make sure that any requirements that dont make it in get follow up tickets
eevvee33Last Friday at 8:01 PM
Would quests reporting be global or local to each discord server?  If a quest was reported for a stop just outside of you discord area, would the bot reject it?  If a quest was reported in an area where two discord servers overlapped, would both discords get the notification or just the reporting one?
IVpipsLast Friday at 8:02 PM
So all pokenav features are "tenanted" by their server, the app will expose it globally
itll work exactly like raids
xxTopCatxxLast Friday at 8:02 PM
The more I think the more I think that could work so well,  if you want to benefit from what Discord does, come and be a part of it instead of reaping the rewards with no contribution :thumbup::skin-tone-2:
IVpipsLast Friday at 8:02 PM
dedeuplication based on the stop, per server
if you want to see multiple servers, look at the app
makes it easier to reason about
plus it makes configuration easier
you dont really want other servers to post into yours
Cupkait 🍰Last Friday at 8:06 PM
I don’t like the idea of limiting the quests to be visible only by those calling out. I don’t want pinap quests called out at all and all that would do is encourage spam callouts when all my server wants to see are the valuable ones.
IVpipsLast Friday at 8:07 PM
My thought on the flow is:

1.) Post a screenshot of the stop, OCR the name
2.) If the message contains any text, check for a known research reward in the name (i.e. a pokemons name, or a berry, pokeball, stardust, etc) and set that as the research reward
3.) If not present the standard 60 second interactive dialogue to select the quest rewards
1+2 allows us to create a siri shortcut for posting quests in one action
also obviously there will be a command to report
"stop" "reward"
IVpipsLast Friday at 8:08 PM
also I'm thinking about grouping by reward in the quest channel(edited)
So "caterpie" would list all the stops that contain caterpie quests for that day
links to the stops like in that embed(edited)
zFatKatLast Friday at 8:09 PM
Grouping would be amazing
IVpipsLast Friday at 8:10 PM
it should be feasible
Cupkait 🍰Last Friday at 8:10 PM
Yes.
zFatKatLast Friday at 8:10 PM
:tada::tada::tada::tada:
I am excited
IVpipsLast Friday at 8:10 PM
false reports ?
is deleting false reports a mod / admin action(edited)
Cupkait 🍰Last Friday at 8:10 PM
Yes or creator
IVpipsLast Friday at 8:11 PM
yeah list-quests or list-research
Cupkait 🍰Last Friday at 8:11 PM
If someone else reports it as false then alert mod?
IVpipsLast Friday at 8:11 PM
can list all the research you are allowed to delete with ids to delete
ok
i'll sort that part out later and think on it a little
xxTopCatxxLast Friday at 8:11 PM
Good point on the pinaps
eevvee33Last Friday at 8:12 PM
I did grouping with my bot reporting.  However, I think that only really works if you batch reports a few times per day.  If you reported immediately there wouldn't really be anything to group.  So, maybe batch reports once every 30, or 60, minutes so that the report can group reports by reward type.
IVpipsLast Friday at 8:12 PM
Couldn't it just build it as it goes
if someone reports "stardust" -- if the community has a stardust message, add it to the message
eevvee33Last Friday at 8:13 PM
Oh, yeah, if it can edit a post, that would work too.
IVpipsLast Friday at 8:13 PM
otherwise create a message
thats what I was thinking
Cupkait 🍰Last Friday at 8:13 PM
Yes that
IVpipsLast Friday at 8:13 PM
One limitation with that is 6000 characters per research reward type
eevvee33Last Friday at 8:14 PM
And make the message "day based".  So, if it's a new day new reports go into a new message and not one from yesterday.
IVpipsLast Friday at 8:15 PM
yeah actually I think I'm going to try to figure out the "clearing" part since everyone needs to clear at midnight on their timezone
or at least the bot needs to be able to create new messages and clear the previous days
SunkastLast Friday at 8:20 PM
i would also want a way to manually clear all tasks in the event they change over for the first of the month or an event which can occur in the middle of the day(edited)
Cupkait 🍰Last Friday at 8:20 PM
^^^^^ yes
zFatKatLast Friday at 8:23 PM
Absolutely
Like a big red button to clear quests lol
Cupkait 🍰Last Friday at 10:01 PM
Would love to see an active raids consolidation similar to how I figure you’re imagining grouping quests. But for tier of raids. :innocent:

**Comment by Rhendak** [comment]
RhendakToday at 3:35 PM
@IVpips to respond on the whole "research functionality" discussion from the other day:

I kinda like the idea you where brainstorming with: 
1) user uploads a screenshot of the pokéstop (or starts with a command $research <pokestop name>
2) user is prompted for the quest reward.
3) bot automatically looks up matching quest to the given quest reports and forms the reported quest
4) any previous uploaded screenshots/issued commands as well as reaction to prompts from the user are deleted to keep a clean channel

I think the above system/idea/process is even better then what I initially suggested

**Comment by dandesousa** [comment]
This is about to be closed with the public beta, all other features should be done as follow up requests. There will be a server announcement with a feature overview and what was included / left out.
